Invoke stubs fail with latest changes

Hi,

with the latest changes added to the invoke-stubs, mypy fails

```py
from invoke import task, Context

@task
def docker_build(context: Context) -> None:
    pass

@task(docker_build)
def docker_push(context: Context) -> None:
    pass
```

```
foo.py:9:2: error: Argument 1 to "__call__" of "Task" has incompatible type "Callable[[Context], None]"; expected "Context"  [arg-type]
    @task(docker_build)
```

As soon as I add any argument to `@task(docker_build)` like `@task(docker_build, name="fooo")` it works. I guess mypy fails to distinguish task overload 1 and 2. Do you have any idea how to solve this?

Hi, the issue is closed by the change of @AlexWaygood , but I found another one

```py
from __future__ import annotations

from typing import Any

from invoke import Context, Task, task

test_task2: Task[..., Any]


@task(pre_ci_test2)
def test_task(context: Context) -> None:
    pass
```

-> `foo.py:10:2: error: Untyped decorator makes function "test_task" untyped  [misc]`

This is Independent of the order of the overloads. In my case I have a factory which creates test_task2 and returns it as `Task[..., Any]`. However, mypy gives up here

--- a/stubs/invoke/@tests/test_cases/check_task.py
+++ b/stubs/invoke/@tests/test_cases/check_task.py
@@ -6,6 +6,7 @@ from invoke import Context, task
 # This snippet is a regression test for #8936
 # ===========================================
 
+
 @task
 def docker_build(context: Context) -> None:
     pass

--- a/stubs/invoke/@tests/test_cases/check_task.py
+++ b/stubs/invoke/@tests/test_cases/check_task.py
@@ -1,0 +1,11 @@
+# pyright: reportUnnecessaryTypeIgnoreComment=true
+
+from invoke import task, Context
+
+@task
+def docker_build(context: Context) -> None:
+    pass
+
+@task(docker_build)
+def docker_push(context: Context) -> None:
+    pass

--- a/stubs/invoke/@tests/test_cases/check_task.py
+++ b/stubs/invoke/@tests/test_cases/check_task.py
@@ -2,6 +2,9 @@
 
 from invoke import Context, task
 
+# ===========================================
+# This snippet is a regression test for #8936
+# ===========================================
 
 @task
 def docker_build(context: Context) -> None:

--- a/stubs/invoke/@tests/test_cases/check_task.py
+++ b/stubs/invoke/@tests/test_cases/check_task.py
@@ -1,10 +1,12 @@
 # pyright: reportUnnecessaryTypeIgnoreComment=true
 
-from invoke import task, Context
+from invoke import Context, task
+
 
 @task
 def docker_build(context: Context) -> None:
     pass
+
 
 @task(docker_build)
 def docker_push(context: Context) -> None:

--- a/stubs/invoke/invoke/tasks.pyi
+++ b/stubs/invoke/invoke/tasks.pyi
@@ -59,8 +59,6 @@ class Task(Generic[_P, _R_co]):
     def get_arguments(self, ignore_unknown_help: bool | None = ...) -> list[Argument]: ...
 
 @overload
-def task(__func: Callable[_P, _R_co]) -> Task[_P, _R_co]: ...
-@overload
 def task(
     *args: Task[..., Any],
     name: str | None = ...,
@@ -93,6 +91,8 @@ def task(
     incrementable: Iterable[str] | None = ...,
     klass: type[_TaskT],
 ) -> Callable[[Callable[..., Any]], _TaskT]: ...
+@overload
+def task(__func: Callable[_P, _R_co]) -> Task[_P, _R_co]: ...
 
 class Call:
     task: Task[..., Any]

